src/test/java/org/junitpioneer/jupiter/FailAtExtensionTests.java

```diff
@@ -0,0 +1,122 @@
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junitpioneer.testkit.ExecutionResults;
+import org.junitpioneer.testkit.PioneerTestKit;
+
+@DisplayName("Tests for the FailAt extension")
+class FailAtExtensionTests {
+
+	@Test
+	@DisplayName("Should not fail test without annotation")
+	void shouldNotFailTestWithoutAnnotation() {
+		final ExecutionResults results = PioneerTestKit.executeTestMethod(FailAtTestCases.class, "testNoAnnotation");
+		assertThat(results).hasSingleStartedTest();
+		assertThat(results).hasSingleSucceededTest();
+		assertThat(results).hasNumberOfSkippedTests(0);
+		assertThat(results).hasNoReportEntries();
+	}
+
+	@Test
+	@DisplayName("Should enable test with unparsable `date`` string")
+	void shouldEnableTestWithUnparsableUntilDateString() {
+		final ExecutionResults results = PioneerTestKit
+				.executeTestMethod(FailAtTestCases.class, "testUnparsableUntilDateString");
+		assertThat(results).hasSingleStartedTest();
+		assertThat(results).hasSingleFailedTest();
+		assertThat(results).hasNoReportEntries();
+	}
+
+	@Test
+	@DisplayName("Should fail test with `date` in the past")
+	void shouldFailTestWithFailAtDateInThePast() {
+		final ExecutionResults results = PioneerTestKit
+				.executeTestMethod(FailAtTestCases.class, "testIsAnnotatedWithDateInThePast");
+		assertThat(results).hasSingleStartedTest();
+		assertThat(results).hasSingleFailedTest();
+		assertThat(results).hasSingleReportEntry().firstValue().contains("is after or on the `date`");
+
+	}
+
+	@Test
+	@DisplayName("Should not fail a test with `date` in the future")
+	void shouldNotFailTestWithFailAtDateInTheFuture() {
+		final ExecutionResults results = PioneerTestKit
+				.executeTestMethod(FailAtTestCases.class, "testIsAnnotatedWithDateInTheFuture");
+		assertThat(results).hasSingleSucceededTest();
+		assertThat(results).hasSingleReportEntry().firstValue().contains("2199-01-01", "did not fails the test");
+	}
+
+	@Test
+	@DisplayName("Should fail nested test with `date` in the past when meta annotated by higher level container")
+	void shouldFailNestedTestWithFailAtDateInThPastWhenMetaAnnotated() {
+		final ExecutionResults results = PioneerTestKit
+				.executeTestMethod(FailAtTestCases.NestedTestCases.class, "shouldRetrieveFromClass");
+		assertThat(results).hasSingleFailedContainer();
+		assertThat(results).hasNumberOfStartedTests(0);
+		assertThat(results).hasSingleReportEntry().firstValue().contains("is after or on the `date`");
+	}
+
+	static class FailAtTestCases {
+
+		@Test
+		void testNoAnnotation() {
+
+		}
+
+		@Test
+		@FailAt(reason = "I don't know how to write dates!", date = "xxxx-yy-zz")
+		void testUnparsableUntilDateString() {
+
+		}
+
+		@Test
+		@FailAt(reason = "Everything was better in the past!", date = "1993-01-01")
+		void testIsAnnotatedWithDateInThePast() {
+
+		}
+
+		// How can we test today? This way gives "attribute must be constant compile error"
```

Comment:
@Michael1993  Do you have an idea? We don't test today in the `@DisabledUntil` extension either so we miss this test edge case there too.

Reply:
We can create a mock `ExecutionContext` and call the extension directly. Not sure if all the orchestration is worth it, but that is one way to do it...

Reply:
> We can create a mock `ExecutionContext` and call the extension directly. Not sure if all the orchestration is worth it, but that is one way to do it...

Decided to trust the JDK guys that they know how to implement `before` and that you will see if my if condition is correct by looking at it during PR review.